components/webui/imports/ui/IngestView/Details.jsx

```diff
@@ -0,0 +1,109 @@
+import dayjs from "dayjs";
+
+import {
+    faChartBar,
+    faClock,
+    faEnvelope,
+    faFileAlt,
+} from "@fortawesome/free-solid-svg-icons";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+
+import {DATETIME_FORMAT_TEMPLATE} from "/imports/utils/datetime";
+
+import Panel from "./Panel";
+
+
+/**
+ * Presents details from the given statistics.
+ *
+ * @param {object} props
+ * @param {CompressionStats} props.stats
+ * @return {React.ReactElement}
+ */
+const Details = ({stats}) => {
+    const {
```

Comment:
Right, except that the definitions written by JSDoc has no runtime perf impact, there is no special reason a class was not created. 

> maintenance overhead of not using classes

On my end, code completion and type navigation do seem to work fine in PyCharm. Is there a specific scenario you are finding the docs hard to maintain? (Depending on what you find, the suggestion below might not apply.

I do agree that in the long term we should rely less on JSDoc for documenting those typing information. Given nowadays TypeScript is becoming more prevalent and people are ditching JSDoc for it, it might worth a try for us to introduce TypeScript into this project, even if we just use it to document the types.